### ISSUE TITLE
Feature: Support for plugins that extend the UI

### ISSUE THREAD

**Issue by 4naesthetic** [klarna-incubator/gram#101]
I would like to be able to write plugins that extend the API and UI together. One example could be a plugin that allows models to be exported in a particular format, which would include a new button or menu item on the front end in addition to new APIs.

Apologies if this is already possible - it wasn't clear to me how to achieve this without modifying the frontend.

**Comment by Tethik** [comment]
You're not wrong, currently this is not possible besides some minor changes from `config/` where you can add menu items. It would be cool to have, but I expect it's pretty hard to implement.

**Comment by Tethik** [comment]
That said, exporting threat models to other formats for downloading is also something that could be implemented in Gram itself as a feature.
